### PR TITLE
Add FAAS OSD PVC check

### PR DESCRIPTION
**Pull request by ebondare**

### CODE REVIEW

**Comment by ebondare** [comment]
Successful test run: https://ocs4-jenkins-csb-odf-qe.apps.ocp-c1.prod.psi.redhat.com/job/qe-deploy-ocs-cluster/25253/

File: ocs_ci/helpers/managed_services.py [hunk 1]
```diff
@@ -500,7 +500,34 @@ def verify_faas_provider_resources():
...
+        assert pvc["spec"]["storageClassName"] == constants.DEFAULT_OCS_STORAGECLASS, (
+            f"Storage class of PVC {pvc['metadata']['name']} is "
+            f"pvc['spec']['storageClassName']. "
+            f"It should be {constants.DEFAULT_OCS_STORAGECLASS}"
+        )
+        defaultsc = OCP(
+            kind=constants.STORAGECLASS,
+            namespace=config.ENV_DATA["cluster_namespace"],
+            resource_name=constants.DEFAULT_OCS_STORAGECLASS,
+        )
```
Comment (yitzhak12) [inline ocs_ci/helpers/managed_services.py:525]: Please let me know if I am wrong, but this should be outside the `for` loop.

File: ocs_ci/helpers/managed_services.py [hunk 2]
```diff
@@ -500,7 +500,34 @@ def verify_faas_provider_resources():
...
+    osdpvcs = OCP(
+        kind=constants.PVC,
+        namespace=config.ENV_DATA["cluster_namespace"],
+        selector=constants.OSD_PVC_GENERIC_LABEL,
+    )
+    for pvc in osdpvcs.get().get("items"):
+        log.info(f"Verifying storageclass of OSD PVC {pvc['metadata']['name']}")
+        assert pvc["spec"]["storageClassName"] == constants.DEFAULT_OCS_STORAGECLASS, (
+            f"Storage class of PVC {pvc['metadata']['name']} is "
+            f"pvc['spec']['storageClassName']. "
```
Comment (yitzhak12) [inline ocs_ci/helpers/managed_services.py:518]: One tiny change: Replace this line with `f"{pvc['spec']['storageClassName']}. "`

File: ocs_ci/helpers/managed_services.py [hunk 3]
```diff
@@ -500,7 +500,34 @@ def verify_faas_provider_resources():
...
+        )
+        defaultsc = OCP(
+            kind=constants.STORAGECLASS,
+            namespace=config.ENV_DATA["cluster_namespace"],
+            resource_name=constants.DEFAULT_OCS_STORAGECLASS,
+        )
+    defaultsc_info = defaultsc.get()
+    assert defaultsc_info["parameters"]["type"] == constants.GP3, (
+        f"Type of OSD PVC's storage class is {defaultsc_info['parameters']['type']}. "
+        f"It should be constants.GP3"
```
Comment (yitzhak12) [inline ocs_ci/helpers/managed_services.py:529]: Replace this line with `f"It should be {constants.GP3}"`

**Review by fbalak — APPROVED** [review]

File: ocs_ci/helpers/managed_services.py (hunk 2 above)
Comment (ebondare) [reply ocs_ci/helpers/managed_services.py]: fixed

File: ocs_ci/helpers/managed_services.py (hunk 3 above)
Comment (ebondare) [reply ocs_ci/helpers/managed_services.py]: fixed

**Comment by ebondare** [comment]
New test run: https://ocs4-jenkins-csb-odf-qe.apps.ocp-c1.prod.psi.redhat.com/job/qe-deploy-ocs-cluster/25336/

**Review by yitzhak12 — APPROVED** [review]

**Pull request merged**